### PR TITLE
Assembles chunked_uploads in workers working directory

### PR DESCRIPTION
**Pull request by dkliban**
fixes: #9550
https://pulp.plan.io/issues/9550

### CODE REVIEW

File: pulpcore/app/tasks/upload.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def commit(upload_id, sha256):
         return
 
     chunks = models.UploadChunk.objects.filter(upload=upload).order_by("offset")
-    with NamedTemporaryFile("ab") as temp_file:
+    with NamedTemporaryFile(mode="ab", dir=".", delete=False) as temp_file:
```
Comment (bmbouter) [inline pulpcore/app/tasks/upload.py:29]: Does this codepath happen inside a path? I'm thinking yes otherwise what would clean it up?

Comment (dkliban) [reply pulpcore/app/tasks/upload.py:29]: inside what path? clean up when?

Comment (dkliban) [reply pulpcore/app/tasks/upload.py:29]: This is a task being executed by a worker. The "." is the worker's directory. NamedTemporaryFile creates a temporary file inside that directory. When the delete is set to True, the file is deleted after it's closed on line 34. In order to keept it around until it is moved into artifact storage we need to set delete to False.

Comment (bmbouter) [reply pulpcore/app/tasks/upload.py:29]: Yeah I meant `s/path/task/`. You answered it, thanks.

**Review by bmbouter — APPROVED** [review]
Thanks!

File: pulpcore/app/tasks/upload.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/app/tasks/upload.py:29]: Why is the file mode "ab" and not "wb"? If we used the default ("w+b") it should even be possible to read without needing to reopen.
Not saying it needs changing. It just sticks out as possibly unnecessary.

**Review by mdellweg — APPROVED** [review]

**Comment by bmbouter** [comment]
I agree w/ you @mdellweg

**Review by dralley — APPROVED** [review]

**Pull request merged**